packages/core/src/auth/sso/ssoAccessTokenProvider.ts

```diff
@@ -761,3 +784,53 @@ type ReAuthStateValue = {
     // the latest reason for why the connection was moved in to a "needs reauth" state
     reAuthReason?: string
 }
+
+/**
+ * Singleton class that manages showing the user a message during
+ * failures that ocurr during SSO disk cache operations.
+ *
+ * Background:
+ * - We need this {@link DiskCacheErrorMessage} specifically as a singleton since we want to ensure
+ *   that only 1 instance of this message appears at a time. There are cases where it shows multiple messages
+ *   if not used as a singleton.
+ */
+class DiskCacheErrorMessage {
```

Comment:
putting things in a `class` can help group related logic until we find a better abstraction. but these classes seem to always have a `instance` singleton, why ?

we have `InactivityMessage` which does something similar. maybe it's time to generalize this concept instead of creating a new class for each message.

Reply:
Reason for the singleton is explained a couple lines above.

As for making a unified class, outside of reusing the generic "showViewLogsMessage", what else in this could you see being generalized? InactivityMessage and this message class both have their own business logic for when to show the message, what the message content is. InactivityMessage has specific conditions to show different messages depending on how much time has passed, and it needs to keep track of how long a user has been inactive for. This DiskCache message class cares about throttling the message since it has certain issues it needs to worry about.

It's not clear to me how to generalize the two, but if you can draw some parallels that will help.